File: pid-backend/src/main/java/cz/fit/cvut/pidbackend/Model/Stop.java
package cz.fit.cvut.pidbackend.Model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "stops")
@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class Stop {

    @Id
    @Column(name = "uid")
    private String id;

    @Column(name = "name", nullable = false)
    private String name;
    @Column(name = "lat")
    private Double lat;
    @Column(name = "lon")
    private Double lon;
    @Column(name = "zone_id")
    private String zoneId;
    @Column(name = "wheelchair")
    private int weelchair;
    @Column(name = "parent_station")
    private String parentStation;
}

File: android-app/app/src/main/java/cvut/fit/pidmobapp/model/Stop.kt
package cvut.fit.pidmobapp.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.SerializedName
import cvut.fit.pidmobapp.R
import cvut.fit.pidmobapp.utils.BaseClusterItem

@Entity
data class Stop(
    @PrimaryKey
    @SerializedName("id")
    val uid: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("lat")
    val lat: Double,
    @SerializedName("lon")
    val lon: Double,
    @SerializedName("zoneId")
    val zoneId: String,
    @SerializedName("weelchair")
    val wheelchair: Int,
    @SerializedName("parentStation")
    val parentStation: String
    ) : BaseClusterItem() {

    override fun getPosition(): LatLng =
            LatLng(lat, lon)

    override fun getTitle(): String =
            name

    override fun getSnippet(): String =
            name

    override val drawable: Int
        get() = R.drawable.ic_baseline_directions_bus_24

    override val color: Int
        get() = R.color.black
}
